From 79da031e9811f3eef34b14cce419be93fea34319 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 13 Jun 2017 18:10:06 +0300
Subject: [PATCH] Run installation commands as shell jobs

This greatly speeds up installation time on multi-core systems.

Upstream-Status: Submitted [https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/39]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 src/fullcolor/Makefile.am |  5 +++--
 src/spinner/Makefile.am   |  7 ++++---
 src/symbolic/Makefile.am  | 11 ++++++-----
 3 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/src/fullcolor/Makefile.am b/src/fullcolor/Makefile.am
index d73529b..b7d0808 100644
--- a/src/fullcolor/Makefile.am
+++ b/src/fullcolor/Makefile.am
@@ -9,9 +9,10 @@ install-data-local:
 		for file in `cd $(top_srcdir)/$(SVGOUTDIR)/$$size && find . -name "*.png"`; do \
 			context="`dirname $$file`"; \
 			$(mkdir_p) $(DESTDIR)$(themedir)/$$size/$$context; \
-			$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file; \
+			$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file & \
 		done; \
-	done;
+	done; \
+	wait
 
 ## FIXME we should add a way to remove links generated by icon mapping
 uninstall-local:
diff --git a/src/spinner/Makefile.am b/src/spinner/Makefile.am
index c14caf6..322dc0d 100644
--- a/src/spinner/Makefile.am
+++ b/src/spinner/Makefile.am
@@ -24,13 +24,14 @@ install-data-local:
 	      for file in `cd $(top_srcdir)/$(SVGOUTDIR)/$$size; find . -name "*.png"`; do \
 		      context="`dirname $$file`"; \
 		      $(mkdir_p) $(DESTDIR)$(themedir)/$$size/$$context; \
-		      $(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file; \
+		      $(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file & \
 	      done; \
 	for file in `cd $(top_srcdir)/$(SVGOUTDIR)/scalable-up-to-32; find . -name "*.svg"`; do \
 		context="`dirname $$file`"; \
 		$(mkdir_p) $(DESTDIR)$(themedir)/scalable-up-to-32/$$context; \
-		$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/scalable-up-to-32/$$file $(DESTDIR)$(themedir)/scalable-up-to-32/$$file; \
-	done
+		$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/scalable-up-to-32/$$file $(DESTDIR)$(themedir)/scalable-up-to-32/$$file & \
+	done; \
+	wait
 
 uninstall-local:
 	      for file in `cd $(top_srcdir)/$(SVGOUTDIR)/scalable-up-to-32; find . -name "*.svg"`; do \
diff --git a/src/symbolic/Makefile.am b/src/symbolic/Makefile.am
index 957c0ee..e1f8818 100644
--- a/src/symbolic/Makefile.am
+++ b/src/symbolic/Makefile.am
@@ -34,18 +34,19 @@ install-data-local:
 		for file in `cd $(top_srcdir)/$(SVGOUTDIR)/$$size; find . -name "*.png"`; do \
 			context="`dirname $$file`"; \
 			$(mkdir_p) $(DESTDIR)$(themedir)/$$size/$$context; \
-			$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file; \
+			$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/$$size/$$file $(DESTDIR)$(themedir)/$$size/$$file & \
 		done; \
-	done
+	done; \
 	for file in `cd $(top_srcdir)/$(SVGOUTDIR)/scalable; find . -name "*.svg"`; do \
 		context="`dirname $$file`"; \
 		$(mkdir_p) $(DESTDIR)$(themedir)/scalable/$$context; \
-		$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/scalable/$$file $(DESTDIR)$(themedir)/scalable/$$file; \
+		$(install_sh_DATA) $(top_srcdir)/$(SVGOUTDIR)/scalable/$$file $(DESTDIR)$(themedir)/scalable/$$file & \
 		for size in $(symbolic_encode_sizes); do \
 			$(mkdir_p) $(DESTDIR)$(themedir)/$$size/$$context; \
-			$(GTK_ENCODE_SYMBOLIC_SVG) $(top_srcdir)/$(SVGOUTDIR)/scalable/$$file $$size -o $(DESTDIR)$(themedir)/$$size/$$context; \
+			$(GTK_ENCODE_SYMBOLIC_SVG) $(top_srcdir)/$(SVGOUTDIR)/scalable/$$file $$size -o $(DESTDIR)$(themedir)/$$size/$$context & \
 		done \
-	done
+	done; \
+	wait
 
 uninstall-local:
 	for file in `cd $(top_srcdir)/$(SVGOUTDIR)/scalable; find . -name "*.svg"`; do \
